app/src/main/java/com/mstudio/superheroarch/presentation/DetailsViewModel.kt

```
@@ -1,40 +1,84 @@
+
+                val episodeCode = episodeEntity.episode // Asumimos que este valor tiene el formato "S01E01"
+
+                val seasonAndEpisode = extractSeasonAndEpisode(episodeCode)
+
+                seasonAndEpisode?.let { (season, episode) ->
+                    fetchTMDBEpisodeDetails(season, episode)
+                } ?: run {
+                    view.showError("Invalid episode format")
+                }
             }
         } catch (e: Exception) {
```

detekt

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)